kore/src/Kore/Repl/Parser.hs

```diff
@@ -52,7 +57,16 @@ import qualified Kore.Log as Log
 import qualified Kore.Log.Registry as Log
 import Kore.Repl.Data
 
-type Parser = Parsec String String
+type Parser = Parsec ReplParseError String
+
+newtype ReplParseError = ReplParseError { unReplParseError :: String }
```

Comment:
What are the benefits of using a newtype with an `IsString` instance here rather than `String` directly?

Reply:
When we use `String` directly, we have to declare an orphan instance.

Reply:
Oh, that's right, we needed the `ShowErrorComponent` instance.